packages/ui-form-field/src/FormFieldLayout/index.tsx

```diff
@@ -184,9 +169,9 @@ class FormFieldLayout extends Component<FormFieldLayoutProps> {
         css={styles?.formFieldLayout}
         style={{ width }}
         aria-describedby={this.hasMessages ? this._messagesId : undefined}
+        aria-labelledby={this.props.label ? this._labelId : undefined}
         ref={this.handleRef}
       >
-        {this.elementType === 'fieldset' && this.renderLegend()}
```

Comment:
The old solution used here a `fieldset` element. SR only reads its `<legend>` if its the first element. But we use a `Grid`, so the solution was to add a `<legend>` as `ScreenReaderContent` and use `aria-hidden` on the visible label.

The new solution uses `aria-labelledby` instead, so there is just one, visible DOM element for the label